pydantic/main.py

```diff
@@ -418,9 +433,25 @@ def __iter__(self) -> 'AnyGenerator':
         """
         yield from self._iter()
 
-    def _iter(self, by_alias: bool = False) -> 'TupleGenerator':
+    def _iter(self, by_alias: bool = False, skip_defaults: bool = True) -> 'TupleGenerator':
         for k, v in self.__values__.items():
-            yield k, self._get_value(v, by_alias=by_alias)
+            yield k, self._get_value(v, by_alias=by_alias, skip_defaults=skip_defaults)
+
+    def _calculate_keys(
+        self, include: Set[str] = None, exclude: Optional[Set[str]] = set(), skip_defaults: bool = False
+    ) -> Set[str]:
+        if skip_defaults:
+            keys = self.__fields_set__.copy()
+        else:
+            keys = set(self.__values__.keys())
+
+        if include:
+            keys &= set(include)
```

Comment:
I doubt we need `set()` here.

Reply:
I was erring on the side of safety, not everyone uses mypy. But happy to remove it.

Reply:
I think this one really can be removed. If someone passes something weird here they should get an error.